client/core/core.go

```diff
@@ -344,10 +344,18 @@ func New(cfg *Config) (*Core, error) {
 	if err != nil {
 		return nil, fmt.Errorf("database initialization error: %v", err)
 	}
+	certMap := make(map[string]string, len(cfg.Certs))
+	for k, path := range cfg.Certs {
+		url, err := url.Parse(k)
+		if err != nil || url.Host == "" {
+			return nil, fmt.Errorf("failed to parse certificate host from %s", k)
+		}
+		certMap[url.Host] = path
```

Comment:
I'm not sure if this is desirable since you can't just use "host:port" in the certs.json file.  I feel like we should be using `net.SplitHostPort` and change the convention.  (https://play.golang.org/p/m_DdQ6nDtDp)